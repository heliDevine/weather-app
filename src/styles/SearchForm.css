@import url('https://fonts.googleapis.com/css2?family=Fira+Sans+Condensed:wght@100;400&display=swap');

.search-form {
    display: flex;
    flex-direction: row;
    flex-grow: inherit;
    padding: 100px, 60px, 60px, 60px;
    margin: 20px;
}

.button {
    size: 100px;
    font-family: 'Fira Sans Condensed', sans-serif;
    font-weight: 400;
    color: red;
}
