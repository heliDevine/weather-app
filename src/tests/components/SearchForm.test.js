import React from 'react';
import SearchForm from '../../components/SearchForm';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

// describe('SearchForm', () => {
//     const validProps = {
//         searchText: () => {},
//         setSearchText: () => {},
//         onSubmit: () => {},
//     };
//     it('renders correctly', () => {
//         const testRenderer = renderer
//             .create(
//                 <SearchForm
//                     searchText={validProps.searchText}
//                     setSearchText={validProps.setSearchText}
//                     onSubmit={validProps.onSubmit}
//                 />,
//             )
//             .toJSON();
//         expect(testRenderer).toMatchSnapshot();
//     });
// });

describe('SearchForm', () => {
    const validProps = {
        searchText: 'Stub description',
        setSearchText: () => {},
        onSubmit: () => {},
    };
    it('renders correctly', () => {
        const { asFragment } = render(
            <SearchForm
                searchText={validProps.searchText}
                setSearchText={validProps.setSearchText}
                onSubmit={validProps.onSubmit}
            />,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
