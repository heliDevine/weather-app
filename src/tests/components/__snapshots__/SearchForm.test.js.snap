// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      id="location-search"
      placeholder="City, Country"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
       Search 
    </button>
  </div>
</DocumentFragment>
`;
